sw/device/lib/dif/dif_uart.c

```diff
@@ -498,25 +496,19 @@ dif_uart_result_t dif_uart_fifo_reset(const dif_uart_t *uart,
       mmio_region_read32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET);
 
   if (reset == kDifUartFifoResetRx || reset == kDifUartFifoResetAll) {
-    reg = bitfield_field32_write(reg,
-                                 (bitfield_field32_t){
-                                     .mask = 0x1,
-                                     .index = UART_FIFO_CTRL_RXRST_BIT,
-                                 },
-                                 1);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_RXRST_BIT, true);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
+    reg = bitfield_bit32_write(reg, UART_FIFO_CTRL_RXRST_BIT, false);
+    mmio_region_write32(uart->params.base_addr, UART_FIFO_CTRL_REG_OFFSET, reg);
```

Comment:
I believe Alphan covers this bellow, but I don't think this second write is necessary, at least this is my interpretation.

> RX fifo reset. Write 1 to the register resets RX_FIFO. Read returns 0

Reply:
I think this is something that have come up in the HW_USAGE_REVIEWED meeting, hopefully @eunchan would be able to clarify this.